docs/design/rosbag2_record_replay_service.md

```diff
@@ -0,0 +1,126 @@
+# Rosbag2 record and replay service
+
+## Context
+
+In [ros2/rosbag2#773](https://github.com/ros2/rosbag2/issues/773), it was recommended that record service requests feature should be added to rosbag2.  
+
+This design describe how to implement this feature based on current architecture and how to extend current command parameter.  
+
+## How to save record service message
+
+Service introspection ([ros2/ros2#1285](https://github.com/ros2/ros2/issues/1285)) had been implemented. All service messages (requests and responses) are published to service event topic which is hidden topic named as `_service_event` under service name (e.g. /add_two_ints/_service_event). So all messages in this topic can be recorded like general topic.
+
+## How to distinguish service event topic in all recorded topics
+
+Service event topic is saved like general topic. But they should use different process after reading from recording file. So they should be distinguished by topic name.
+
+- General topic  
+  
+  [NameSpace/]TopicName
+
+- Hidden topic  
+  
+  [NameSpace/]_TopicName
+
+- Service event topic
+  
+  [NameSpace/]ServiceName/_service_event
+
+The following criteria will be used to determine a service event topic:
+1. The topic ends with `/_service_event`
+2. The topic type name matches  `*/srv/*_Event` (e.g. `example_interfaces/srv/AddTwoInts_Event`)
+
+### Expand the 'record' command
+
+Add 5 parameters.
+
+- `--services ServiceName1 [ServiceName2 ...]`
+  
+    Specify which services will be recorded. User must specify at least one service.
+
+- `--all-topics`
+
+    Record all topics. Hidden topics including service event topics are excluded.
+
+- `--all-services`
+
+    Record all service event topics.
+
+- `--exclude-topics`
+
+    Rename from `--exclude`. Exclude topics containing provided regular expression.
+
+- `--exclude-services`
+
+    Exclude services containing provided regular expression.
+
+The description of the relevant parameter behavior.
+
+| Parameter | Description |
+| :-- | :--|
+|--services ServiceName1 [ServiceName2 ...] | Record services (service event topics) with specified service names. |
+| --all | Record all topics and service event topics. Other hidden topics are excluded. |
+| --all-topics | Record all topics. Hidden topics including service event topics are excluded. |
+| --all-services | Only record all service event topics. |
+| --exclude-topics | Exclude topics containing provided regular expression. |
```

Comment:
@Barry-Xu-2018 Please add one more row about 
```
|--include-hidden-topics| Record all hidden topics. Include service event topic. |
```
To avoid confusion and explicitly define expectations.

Reply:
Okay. I supplement the description of `--include-hidden-topics`.